### PR TITLE
[SERVICE-462] Intent resolver UI

### PR DESCRIPTION
**Pull request by rypete**
Updates the resolver UI to match the spec provided.  No functionality changes are made apart from the 'always remember' select box, which is missing from the spec.

### CODE REVIEW

File: res/provider/ui/selector.html
Comment (tomrobertsOF) [inline res/provider/ui/selector.html:8]: RIP w3-css

File: src/provider/common/Injectables.ts
Comment (tomrobertsOF) [inline src/provider/common/Injectables.ts:14]: Any particular reason for this rename?

File: src/provider/controller/ResolverHandler.ts
Comment (tomrobertsOF) [inline src/provider/controller/ResolverHandler.ts:32]: If we're removing the defaults stuff from the UI does it make sense to also cleanup all of the dafaults parts in the provider? (more for @pjbroadbent )

File: res/provider/ui/resolver/css/resolver.css
Comment (tomrobertsOF) [inline res/provider/ui/resolver/css/resolver.css]: To match what we're doing with other react stuff, shouldn't the css for a component be in a `ComponentName.css` file alongside the component's `.tsx` file and bundled in?

Comment (rypete) [reply res/provider/ui/resolver/css/resolver.css]: Yes, you are correct.  However, there is a current problem with the akkurat ttf font and webpack so we had to do it the 'old school' way.

File: src/provider/common/Injectables.ts
Comment (rypete) [reply src/provider/common/Injectables.ts:14]: Its no longer called selector

File: res/provider/ui/resolver/css/resolver.css
Comment (tomrobertsOF) [reply res/provider/ui/resolver/css/resolver.css]: Ok, fair enough. Are we planning on resolving this at any point, and if so can we make a JIRA to track.

File: src/provider/controller/ResolverHandler.ts
Comment (rypete) [reply src/provider/controller/ResolverHandler.ts:32]: Can you point to the parts you are referring to?

File: src/provider/common/Injectables.ts
Comment (tomrobertsOF) [reply src/provider/common/Injectables.ts:14]: So that's a no I guess

File: src/provider/controller/ResolverHandler.ts
Comment (tomrobertsOF) [reply src/provider/controller/ResolverHandler.ts:32]: Oh, guess that all got removed during the refactor without me noticing. Never mind then.

**Review by tomrobertsOF — APPROVED** [review]
Approved assuming the final design has been signed off by whoever should be approving that kind of thing.

File: res/provider/ui/resolver/css/resolver.css
Comment (rypete) [reply res/provider/ui/resolver/css/resolver.css]: Yes but it will need to be a service tooling fix.  We have a JIRA already for it: SERVICE-527

Comment (tomrobertsOF) [reply res/provider/ui/resolver/css/resolver.css]: 👍

File: src/provider/common/Injectables.ts
Comment (rypete) [reply src/provider/common/Injectables.ts:14]: Sorry let me be more specific.  The JIRA refers to the UI as the 'resolver'.  It was coded using this in mind.  Speaking to @pjbroadbent it sounds like selector was used previously as there an existing resolver component which is no more.  The selector was not a great name so I think it is a welcomed change.

Comment (tomrobertsOF) [reply src/provider/common/Injectables.ts:14]: Fair enough. Thanks!

Comment (pjbroadbent) [reply src/provider/common/Injectables.ts:14]: The "selector" name was to avoid confusion with the `resolveIntent` API's, which returned an array of applications and didn't actually use the UI.

Outside of the codebase, the UI has always been referred to as the 'resolver' though, and since the "resolve" API's were renamed, it made sense to also rename the UI implementation to match it's commonly-understood name.

File: res/provider/ui/resolver/css/resolver.css
Comment (pjbroadbent) [reply res/provider/ui/resolver/css/resolver.css]: Can we keep the react-related css in the previous setup, and have this file only include the font stuff?

File: res/provider/ui/resolver/css/resolver.css
Comment (pjbroadbent) [inline res/provider/ui/resolver/css/resolver.css]: Can this padding be on the container, rather than each of the components?

Comment (rypete) [reply res/provider/ui/resolver/css/resolver.css]: No, the bars for each app need to be 100% width (of the window), otherwise you will see the borders being cut at the beginning of the padding.

Comment (rypete) [reply res/provider/ui/resolver/css/resolver.css]: As discussed, no it cant as it will distort the full width elements

**Comment by ottah** [comment]
[Exit icon](https://raw.githubusercontent.com/HadoukenIO/fdc3-service/b68391382b44ab6664a97bd0856eff8fe3490610/res/provider/ui/resolver/assets/exit.png) is 1120x1120 pixels.

File: src/provider/view/components/Resolver.tsx
Comment (ottah) [inline src/provider/view/components/Resolver.tsx]: More a styling gripe, but props should be `onVerb` and the function should be `handleVerb`.

`... onOpen={handleOpen} ...`

Comment (rypete) [reply src/provider/view/components/Resolver.tsx]: Fixed

**Comment by pjbroadbent** [comment]
Looks good, but some possible incoming design tweaks. Will hold this here for a bit whilst conversations progress.

**Review by pjbroadbent — APPROVED** [review]

**Pull request merged**
